redhat-developer/developers.redhat.com
# [RHDENG-2753] Integrate SSO for Drupal Admin login

This PR introduces the following changes to Drupal:

- Enables SSO login for admin accounts using Keycloak
  - The Keycloak instance can be set on a per-environment basis
  - The Keycloak configuration is set using config overrides, hence why you'll see `Set with config override: check rhd.settings.yml` in the `keycloak.settings.yml`
- Makes changes to the Drupal login page so that login failures are displayed to users for immediate feedback
- Enables the `session_limit` module and limits users to `3` active sessions at any one time

This PR does not limit access to the current Drupal login form, so current logins will continue to work in combination with the SSO. This is expected for now.

This PR has a dependency on #2796 . The work in #2796 is included in this PR because without it, SSO integration does not work and cannot be tested.

### JIRA Issue Link
* https://issues.jboss.org/browse/RHDENG-2753

### Verification Process

* The build should go green
* Navigate into Drupal and ensure that there are no hanging configuration changes to import
* Visit `/user/login` and click on the `Login with your Developer Account` button.
  * Log in using your developers.stage.redhat.com Keycloak account
  * **The email address of the account in the developers.stage.redhat.com Keycloak must match the email address of your account in Druipal**
    * e.g. my drupal account name is `rblake` with an email address of `rblake@redhat.com`. My SSO account shows `rblake@redhat.com` in the `Email` field of the account details. You can check your SSO account [here](https://developers.stage.redhat.com/auth/realms/rhd/account/)
  * Browse around Drupal using your SSO login and ensure you can perform typical content creation actions e.g. edit a page
  * Log out of Drupal and then ensure you can still log in using the default log in mechanism and not the SSO login.
